src/eloelo/elodisco/hero_pool_generators.rs

```diff
@@ -0,0 +1,668 @@
+    }
+
+    fn player_j(pool: &Vec<Hero>) -> (PlayerInfo, Vec<Hero>) {
+        (
+            PlayerInfo {
+                name: DiscordUsername::from("j".to_string()),
+                elo: 1000,
+                is_radiant: true,
+                number_of_heroes_shown: 2,
+            },
+            pool.clone(),
+        )
+    }
+
+    fn player_bixkog(pool: &Vec<Hero>) -> (PlayerInfo, Vec<Hero>) {
+        (
+            PlayerInfo {
+                name: DiscordUsername::from("bixkog".to_string()),
+                elo: 1000,
+                is_radiant: false,
+                number_of_heroes_shown: 1,
+            },
+            pool.clone(),
+        )
+    }
+
+    fn player_dragon(pool: &Vec<Hero>) -> (PlayerInfo, Vec<Hero>) {
+        (
+            PlayerInfo {
+                name: DiscordUsername::from("dragon".to_string()),
+                elo: 100,
+                is_radiant: false,
+                number_of_heroes_shown: 2,
+            },
+            pool.clone(),
+        )
+    }
+
+    fn player_goovie(pool: &Vec<Hero>) -> (PlayerInfo, Vec<Hero>) {
+        (
+            PlayerInfo {
+                name: DiscordUsername::from("goovie".to_string()),
+                elo: 100,
+                is_radiant: true,
+                number_of_heroes_shown: 1,
+            },
+            pool.clone(),
+        )
+    }
+
+    fn default_players(pool: Vec<Hero>) -> Vec<(PlayerInfo, Vec<Hero>)> {
+        vec![
+            player_j(&pool),
+            player_bixkog(&pool),
+            player_dragon(&pool),
+            player_goovie(&pool),
+        ]
+    }
+
+    fn few_players(pool: Vec<Hero>) -> Vec<(PlayerInfo, Vec<Hero>)> {
+        vec![player_bixkog(&pool), player_goovie(&pool)]
+    }
+
+    fn no_duplicates(assignments: &HashMap<PlayerInfo, Vec<Hero>>) -> bool {
+        let total_len = assignments.values().flatten().count();
+        let unique_len = assignments.values().flatten().unique().count();
+        total_len == unique_len
+    }
+
+    fn assignment_in(assignment: &Vec<Hero>, set: &HashSet<Hero>) -> bool {
+        set.is_superset(&assignment.iter().cloned().collect())
+    }
+
+    // every HeroPoolGenerator should pass this test
+    #[test]
+    fn test_base_random() {
+        for _ in 0..N {
+            let players = default_players(default_hero_pool());
+            let mut random_assign = RandomHeroPool::default();
+            let players_assignement = random_assign.assign_heroes(players);
+            assert!(no_duplicates(&players_assignement));
+            for (player, assignement) in players_assignement {
+                println!("{}: {:?}", player.name, assignement);
+                assert_eq!(player.number_of_heroes_shown as usize, assignement.len());
+                assert!(assignment_in(
+                    &assignement,
+                    &default_hero_pool().into_iter().collect()
+                ));
+            }
+            println!("{:?}", random_assign.taken);
```

Comment:
Are these prints necessary? If they only print when the test fails then I guess it's ok to have them, but if they print every time then I'd prefer to remove.